packages/mgt-components/src/components/mgt-picker/mgt-picker.ts

```diff
@@ -0,0 +1,324 @@
+
+  /**
+   * Invalidation period of the cache for the responses in milliseconds
+   *
+   * @type {number}
+   * @memberof MgtPicker
+   */
+  @property({
+    attribute: 'cache-invalidation-period',
+    type: Number
+  })
+  public cacheInvalidationPeriod: number = 0;
+
+  private isRefreshing: boolean;
+
+  private _placeholder: string;
+  private _entityType: string;
+  private _keyName: string;
+
+  @state() private response: any[];
+
+  constructor() {
+    super();
+    this._placeholder = this.strings.comboboxPlaceholder;
+    this._entityType = null;
+    this._keyName = null;
+    this.isRefreshing = false;
+  }
+
+  /**
+   * Refresh the data
+   *
+   * @param {boolean} [hardRefresh=false]
+   * if false (default), the component will only update if the data changed
+   * if true, the data will be first cleared and reloaded completely
+   * @memberof MgtPicker
+   */
+  public refresh(hardRefresh = false) {
+    this.isRefreshing = true;
+    if (hardRefresh) {
+      this.clearState();
+    }
+    this.requestStateUpdate(hardRefresh);
+  }
+
+  /**
+   * Clears the state of the component
+   *
+   * @protected
+   * @memberof MgtPicker
+   */
+  protected clearState(): void {
+    this.response = null;
+  }
+
+  /**
+   * Invoked on each update to perform rendering the picker. This method must return
+   * a lit-html TemplateResult. Setting properties inside this method will *not*
+   * trigger the element to update.
+   */
+  public render() {
+    if (this.isLoadingState && !this.response) {
+      return this.renderTemplate('loading', null);
+    } else if (this.error) {
+      return this.renderTemplate('error', this.error ? this.error : null);
+    }
+
+    return this.response?.length > 0 ? this.renderPicker() : this.renderGet();
+  }
+
+  /**
+   * Render picker.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtPicker
+   */
+  protected renderPicker(): TemplateResult {
+    return html`
+      <fluent-combobox id="combobox" autocomplete="list" placeholder=${this.placeholder}>
+        ${this.response.map(
+          item => html`
+          <fluent-option value=${item.id} @click=${(e: Event) => this.handleClick(e, item)}> ${
+            item[this.keyName]
+          } </fluent-option>`
+        )}
+      </fluent-combobox>
+     `;
+  }
+
+  /**
+   * Render picker.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtPicker
+   */
+  protected renderGet(): TemplateResult {
+    return this.cacheEnabled
+      ? html`
+      <mgt-get 
+        resource=${this.resource}
+        version=${this.version} 
+        scopes=${this.scopes} 
+        max-pages=${this.maxPages} 
+        cache-enabled=${this.cacheEnabled}
+        cache-invalidation-period=${this.cacheInvalidationPeriod}>
+      </mgt-get>`
+      : html`
+      <mgt-get 
+        resource=${this.resource}
+        version=${this.version} 
+        scopes=${this.scopes}
+        max-pages=${this.maxPages}>`;
```

Comment:
```suggestion
    return html`
    <mgt-get 
      resource=${this.resource}
      version=${this.version} 
      scopes=${this.scopes} 
      max-pages=${this.maxPages} 
      ?cache-enabled="${this.cacheEnabled}"
      ?cache-invalidation-period="${this.cacheInvalidationPeriod}">
    </mgt-get>`;
```

Use a `?` before a property to bind it to a boolean.